### PR TITLE
Enable `take` and `take_along_axis` in nvfuser executor

### PR DESCRIPTION
**Pull request by crcrpar**
## What does this PR do?

Let nvfuser executor run `take` and `take_along_axis`.
Also, remove the skipif decorator as the min nvfuser version is now 0.2.8 > 0.0.3 as per https://github.com/Lightning-AI/lightning-thunder/blob/5c0127fd7bd11fbe25858f72097e1f12af37adef/thunder/executors/nvfuserex.py#L32-L33

rel:
- https://github.com/NVIDIA/Fuser/issues/4414
- https://github.com/NVIDIA/Fuser/issues/4415

cc @mruberry @tfogal

### CODE REVIEW

**Comment by crcrpar** [comment]
at glance I cannot tell how the change causes the failures but it seems that this change does causes them... :thinking:

**Comment by riccardofelluga** [comment]
One of the repros from nvfuser is this:

```python
# CUDA devices:
#  0: NVIDIA GeForce RTX 3090
#  1: NVIDIA GeForce RTX 3090
# torch version: 2.7.0+cu126
# cuda version: 12.6
# nvfuser version: 0.2.27+gita8ad0bc
import torch
from nvfuser import FusionDefinition, DataType

def nvfuser_incomplete_fusion(fd : FusionDefinition) -> None :
    T0 = fd.define_tensor(shape=[1, 2, 0, 3], contiguity=[None, False, True, True], dtype=DataType.Float, is_cpu=False, stride_order=[3, 2, 1, 0])
    T1 = fd.ops.permute(T0, dims=[1, 2, 0, 3])
    S2 = fd.define_scalar(0, dtype=DataType.Int)
    S3 = fd.define_scalar(0, dtype=DataType.Int)
    S4 = fd.define_scalar(1, dtype=DataType.Int)
    T5 = fd.ops.iota(S2, S3, S4, dtype=DataType.Int)
    S6 = fd.define_scalar(0, dtype=DataType.Int)
    S7 = fd.define_scalar(1, dtype=DataType.Int)
    S8 = fd.define_scalar(1, dtype=DataType.Int)
    T9 = fd.ops.iota(S6, S7, S8, dtype=DataType.Int)
    T14 = fd.ops.reshape(T1, new_shape=[2, 0, 3])
    S15 = fd.define_scalar(0, dtype=DataType.Int)
    T16 = fd.ops.lt(T5, S15)
    S17 = fd.define_scalar(3, dtype=DataType.Int)
    S18 = fd.define_scalar(0, dtype=DataType.Int)
    T19 = fd.ops.where(T16, S17, S18)
    T20 = fd.ops.cast(T19, dtype=DataType.Int)
    T21 = fd.ops.add(T5, T20)
    S22 = fd.define_scalar(0, dtype=DataType.Int)
    T23 = fd.ops.lt(T9, S22)
    S24 = fd.define_scalar(1, dtype=DataType.Int)
    S25 = fd.define_scalar(0, dtype=DataType.Int)
    T26 = fd.ops.where(T23, S24, S25)
    T27 = fd.ops.cast(T26, dtype=DataType.Int)
    T28 = fd.ops.add(T9, T27)
    S29 = fd.define_scalar(3, dtype=DataType.Int)
    T30 = fd.ops.mul(T28, S29)
    T31 = fd.ops.add(T21, T30)
    T32 = fd.ops.index_select(T14, T31, dim=2)
    T33 = fd.ops.squeeze(T32, dims=[2], squeeze_expanded=False)
    fd.add_output(T33)

with FusionDefinition() as fd:
    nvfuser_fusion_idNone(fd)
```

Do you have the same version of nvfuser also locally?

**Comment by jacobhinkle** [comment]
I believe the `squeeze` might get introduced here: https://github.com/Lightning-AI/lightning-thunder/blob/21667f5ae95cb8fa23edbe7438a4a27983ee87fd/thunder/clang/__init__.py#L933-L936

The above indexing method is called in the diagonal clang op here https://github.com/Lightning-AI/lightning-thunder/blob/21667f5ae95cb8fa23edbe7438a4a27983ee87fd/thunder/clang/__init__.py#L383-L398

**Comment by crcrpar** [comment]
@t-vi @lantiga @mruberry As I think this is ready, could you review this one?

**Comment by IvanYashchuk** [comment]
`take` operation is useful for MoE layer computation using group gemm to shuffle rows for per-expert sorting and grouping.

@t-vi, could you please merge this PR?

**Review by t-vi — COMMENTED** [review]
Thank you @crcrpar  @jacobhinkle @riccardofelluga @IvanYashchuk

**Review by t-vi — APPROVED** [review]
